src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -328,9 +291,39 @@ protected async Task UnregisterTimerAsync(IActorTimer timer)
         protected async Task UnregisterTimerAsync(string timerName)
         {
             await ActorRuntime.DaprInteractor.UnregisterTimerAsync(this.actorTypeName, this.Id.ToString(), timerName);
-            if (this.timers.ContainsKey(timerName))
+        }
+
+        private void ValidateTimerCallback(string callback)
+        {
+            var actorTypeName = this.Host.ActorTypeInfo.ActorTypeName;
+            var actorType = this.Host.ActorTypeInfo.ImplementationType;
+
+            // GetMethod will throw an AmbiguousMatchException if more than one methods are found with the same name
+            var methodInfo = actorType.GetMethod(callback, BindingFlags.Public | BindingFlags.Instance);
+
+            // Check if the method exists
+            if (methodInfo == null)
+            {
+                throw new ArgumentException($"Timer callback method: {callback} does not exist in the Actor class: {actorTypeName}");
```

Comment:
The other possibility is that it's not a public instance method.

Reply:
I changed the binding flags to look for static and private methods as well which makes private method as a valid case